lib/lte_link_control/lte_lc.c

```diff
@@ -1565,6 +1567,162 @@ int lte_lc_modem_events_disable(void)
 	return at_cmd_write(AT_MDMEV_DISABLE, NULL, 0, NULL);
 }
 
+int lte_lc_periodic_search_set(const struct lte_lc_periodic_search_cfg *const cfg)
+{
+	int err;
+	char pattern_buf[4][40];
+
+	if (!cfg || (cfg->pattern_count == 0) || (cfg->pattern_count > 4)) {
+		return -EINVAL;
+	}
+
```

Comment:
Should there be a check here in case value 1 is not set when specifying LTE_LC_PERIODIC_SEARCH_PATTERN_TABLE. So that we can fail faster and provide a reason why.